ihontaryk/homework_2/three_words.py

```diff
@@ -0,0 +1,29 @@
+def three_words(text):
+    """
+    three_words function
+    checks if the string contains three words in succession
+    """
+
+    if type(text) is not str:
+        raise TypeError
+
+    if text == '':
+        raise ValueError
+
+    elements = text.split()
+
+    result = []
+
+    for element in elements:
+        if element.isalpha():
+            result.append(element)
+        else:
+            result = []
+
+        if len(result) >= 3:
+            return True
+        else:
+            continue
+
+    return False
+
```

Comment:
try additional solution using slices:


```
for i in range(len(elements) - 2):
    words = elements[i: i + 2]


```

Reply:
and also like this:

slice_1 = elements[::3]
slice_2 = elements[1::3]
slice_3 = elements[2::3]

for words in zip(slice_1, slice_2, slice_3):
    ...